app/src/main/java/com/jetbrains/handson/mpp/mobile/MainActivity.kt

```diff
@@ -12,9 +17,50 @@ class MainActivity : AppCompatActivity(), ApplicationContract.View {
 
         val presenter = ApplicationPresenter()
         presenter.onViewTaken(this)
+
+        val button = findViewById<Button>(R.id.button)
+        button.setOnClickListener {
+
+            val departureSpinner: Spinner = findViewById(R.id.departure_spinner)
+            val departureStation = departureSpinner.selectedItem.toString()
+            val arrivalSpinner: Spinner = findViewById(R.id.arrival_spinner)
+            val arrivalStation = arrivalSpinner.selectedItem.toString()
+
+            val openURL = Intent(Intent.ACTION_VIEW)
+            val linkToOpen = "https://www.lner.co.uk/travel-information/travelling-now/live-train-times/depart/$departureStation/$arrivalStation/#LiveDepResults"
+            openURL.data = Uri.parse(linkToOpen)
+            startActivity(openURL)
+
+            presenter.onButtonTapped(departureStation, arrivalStation, this)
+        }
     }
 
     override fun setLabel(text: String) {
         findViewById<TextView>(R.id.main_text).text = text
     }
+
+    override fun setDepartureDropdown(stationList: List<String>){
```

Comment:
Minor but there should be a space between the closing parentheses and the opening brace